packages/react-components/react-charts-preview/library/src/components/CommonComponents/PopoverRTL.test.tsx

```diff
@@ -0,0 +1,129 @@
+import * as React from 'react';
+import PopoverComponent from './Popover';
+import { resetIds } from '../../Utilities';
+import { getByClass } from '../../utilities/TestUtility.test';
+import { act, render, screen } from '@testing-library/react';
+import { axe, toHaveNoViolations } from 'jest-axe';
+
+expect.extend(toHaveNoViolations);
+
+function sharedBeforeEach() {
+  resetIds();
+}
+
+describe('ChartHoverCard', () => {
+  test('renders the popover component', () => {
+    const { container } = render(<PopoverComponent isPopoverOpen={true} />);
+
+    const chartHoverCardElement = getByClass(container, /calloutContentRoot/);
```

Comment:
react testing library discourages to use class to fetch DOM elements. Don't use it.

Reply:
The DOM structure of the popover element primarily contains class attributes. Therefore, we are testing the popover element using RTL by referencing the class name.

![Screenshot 2024-09-09 165933](https://github.com/user-attachments/assets/2adcea4a-833b-4f11-96df-299c8034f736)

Reply:
I have changed 'getByClass' with 'getById' and 'getByText' in some places where it could possible but there are some remaining test cases where it is hard replacing the 'getByClass' method. I have pushed the latest change in this PR. Please review the new changes.

Reply:
In "PopoverRTL.test.tsx," there are 14 test cases. I have updated 10 test cases by modifying the 'getByClass' with 'getById' and 'getByText.' However, 4 test cases cannot be modified because the DOM structure relies solely on the class attribute. 

Should we delete those 4 test cases from the "PopoverRTL.test.tsx" file, or should we leave them as they are?